app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,63 @@
+class AlbumsController < ApplicationController
+
+  def index
+    @albums = Album.all
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params)
+    @album.update(:rank => 0)
+    if @album.save
+      redirect_to album_path(@album.id)
+    else
+      render "new"
+    end
+
+  end
+
+  def show
+    @album = Album.find(params[:id])
+  end
+
+  def edit
+    @album = Album.find(params[:id])
+  end
+
+  def update
+    @album = Album.find(params[:id])
+    @album.update_attributes(album_params)
+    if @album.save
+      redirect_to album_path(params[:id])
+    else
+      render "edit"
```

Comment:
Another option is to redirect_to the edit action. There are some tradeoffs, though, like not being able to show the updated attributes. Personally, I like your way better.